oracle/provider/fin.go

```diff
@@ -0,0 +1,309 @@
+package provider
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ojo-network/price-feeder/oracle/types"
+)
+
+const (
+	finRestURL               = "https://api.kujira.app"
+	finPairsEndpoint         = "/api/coingecko/pairs"
+	finTickersEndpoint       = "/api/coingecko/tickers"
+	finCandlesEndpoint       = "/api/trades/candles"
+	finCandleBinSizeMinutes  = 5
+	finCandleWindowSizeHours = 240
+)
+
+var _ Provider = (*FinProvider)(nil)
```

Comment:
I think you need to add a `StartConnections` method performing a noop to fully implement the provider interface